bin/cmsl1t

```diff
@@ -42,7 +42,7 @@ def process_tuples(config, nevents, analyzers):
     from cmsl1t.playground.eventreader import EventReader
     reader = EventReader(input_files, events=nevents)
 
-    results = [analyzer.prepare_for_events(reader) for analyzer in analyzers]
+    results = [analyzer.prepare_for_events(reader, puBins) for analyzer in analyzers]
```

Comment:
I'm not sure this is the right solution to this.  For example, I don't think all analyzers will want to know pile-up binning.  And as a generic solution, it doesn't scale in a good way when we want to pass other bits of config info through to an analyzer.  The `load_analyzers` function already provides something like this behaviour, providing a way to pass parameters through to the analyzer init method.  You should see if you can make use of that method and if not see what we need to do to adapt it.

Reply:
I can't see where load_analyzer passes any options tbh. Could use kwargs?

Passing the PU bins from the yaml is something we definitely could do with from the get-go, changing PU bins without changing the code will be needed for various studies.

Reply:
This line:
https://github.com/cms-l1t-offline/cms-l1t-analysis/blob/master/bin/cmsl1t#L178
is where we pass parameters to the Analyzer's init method.  `params` is a dict which is unpacked and passes as parameters to the method.  It's contents is taken from the parameters given for this analyzer in the config file itself.  This might not be the best way to do things, in which case we can adapt it, but I don't think what we have in this edit is an improvement.